crates/ruff_python_parser/src/error.rs

```diff
@@ -667,6 +694,7 @@ impl Display for UnsupportedSyntaxError {
             UnsupportedSyntaxErrorKind::StarExpressionInIndex => {
                 "Cannot use star expression in index"
             }
+            UnsupportedSyntaxErrorKind::StarAnnotation => "Cannot use star annotation",
```

Comment:
Technically, it's an "annotation expression" but as a user facing message I think either "star annotation" or "star annotation expression" is fine.